test/es6/ES6Symbol.js

```diff
@@ -972,6 +972,24 @@ var tests = [
             assert.throws(function () { o[Symbol.iterator](); }, TypeError, "Calling non-existent built-in symbol property with description fails", "Object doesn't support property or method 'Symbol(Symbol.iterator)'");
         }
     },
+    {
+        name: "Symbol equality logic",
+        body() {
+            assert.isTrue(Symbol.toPrimitive != 1);
+            assert.isTrue(Symbol.toPrimitive != NaN);
+            
+            var valueOfCalled = false;
+            var a = Symbol('f');
+            var b = {
+                valueOf : function() {
+                    valueOfCalled = true;
+                    return a;
+                }
+            };
+            assert.isTrue(a == b);

```

Comment:
I didn't understand? Why 
```Object(a) == a
```? 
```typeof Object(a)``` is actually ```object```. 
For Chrome as well ```Object(a) == a``` is ```false```

---
In reply to: [99263716](https://github.com/Microsoft/ChakraCore/pull/2461#discussion_r99263716) [](ancestors = 99263716)